apiserver/worker/compiler.py

```diff
@@ -675,6 +675,10 @@ def compile_anything(bot_dir, installTimeLimit=600, timelimit=600, max_error_len
                 print("error")
                 print(e)
 
+            # let's assume that anyone using cmake is using it for C++
+            if name == "CMake":
+                name = "C++"
+
```

Comment:
Instead of a second special casing for cmake could it be combined into the extra cmake conditional at line 610, just change the lang.name there?

Reply:
Unfortunately no, because compilation happens right between these two places.

Reply:
(And you need CMake for compilation.)

Reply:
Though hopefully I'm misunderstanding what you are proposing and it _can_ be made nicer.